### ISSUE TITLE
Unable to create a resources group on azure

### ISSUE THREAD

**Issue by mbajur** [deis/deis#4803]
Hello,

I am trying to create my first azure 3-nodes deis cluster using an official tutorial (http://docs.deis.io/en/latest/installing_deis/azure/). However, i'm not able to do that using steps provided - there is one step that makes trouble: sshKeyData. I've searched for some other tutorials and found that all of them states the same so it must be something on my side. I just can't understand what i'm doing wrong so maybe someone here will help me.

Anyway - all the tutorials states that i need to, more or less, make the follwing (i've skipped the rest for simplicity's sake)

- generate a new ssh key `~/.ssh/deis(.pub)`
- add a content of `~/.ssh/deis` to `parameters.json` to `sshKeyData` value.
- run `azure group create (...)`
- add deis key `ssh-add -K ~/.ssh/deis`

However, for some reason, azure wont let me put contents of `~/.ssh/deis` as sshKeyData (i've removed the beginning and end lines and all the new-lines from it). When i try to run `azure group create` with `deis` key in `parameters.json`, i can see in the management portal that provisioning has failed because:

```
{"error":{"code":"InvalidParameter","target":"linuxConfiguration.ssh.publicKeys.keyData","message":"The value of parameter linuxConfiguration.ssh.publicKeys.keyData is invalid."}}
```

I'm able to finalize a provisioning by providing `deis.pub` instead of `deis` contents but it seems not to be the way to go and it breaks deisctl later anyway so i'm not gonna write about that (or should i?).

my `sshKeyData` looks like that (the `(...)` in the middle is intended):

```
"sshKeyData": {
    "value": "MIIEpAIBAAKCAQEAlyKPSn1R9BZiMnUbaZpk07BmiHGk0X5c6hqeyIeex7CtReoplIJ3Bi3CC/2irhCZ8OgS3ELFmQbWFXf5SwrnO27YdlSyXHLtIRvvse/74tLv3CeCM5EJgPq8ze0zfkwPEGHmi6sbaC8lBM021amgLOn6OfQp8dQHGX2HOhN1KoUIaRAHPH4nICSAD1nPKH45dXL5R9qIz8pDOcY8b5QnwrpP7U/GSje9WKRHkXfbWejZWQ6rpSHYgTyAtbV0d0Ulp25nUkRGQvc+pObhVWsnsCpCXgOjt8csrQGvLTJNW6yYVNci29j5mDr8/k0ro/omncziSG6HX7DlQ6wfIqU2MQIDAQABAoIBAD3lUXnYSTAF2sF/ZHKjYJexrsxCfpWtgq/9pvnJUSXHHa98bvr7U3OcWdQ/U9ilw+X9pXu/fN1nB1ZThC1PkdOtWclcdo+WIbdlkYz2E9qZONKY+6HSzGmeRqSveF9JFwJGlfytXtQUOrz0rnhit86tJGWHsYXwp9ewqSkHvvxiTiod8QFEkO9v64whY+vIdIaR8ZnaffOIVt+aLUoDz2In4uhx7zU1CHkR/4SORSIrLD+ET4qWkSCUL0/mzj7FgESSo0ZRK/z+K9RCesGHUzxGM7rFSXR4TnzzOR/Aao2YdE2cyv0DaWpkA8LQaNrLnhHcCtkseHbT3hcRZTViwoECgYEAxLqDZLdLWsLMSONAq07CScb+gzCG5K8XhT5N6YzmOEzDcc45Nmxgb56hm2SH8giZ0AqQorQ1SdzVRbsejnU2v(...)yr52sWfydOFZfTQWj9xsUtkRSUvSoCHE6GHZBpVGUiaduhUx6wXE6nbuIRDqS4VKZRuReZkY6mbGDhXPXn7Q4aMs83s8P+q4MsILQKBgQCi2keSKwlzbtaoFjK5AK6cnRn7abY3BZpD9sdT2cd1KZERPAMgGj2jw6ayT/8lx69DnqClzhhhWeyr5lsE9thUSNwHJoBodvEVHvCOmv5K80l3BmS/jROJHo1ljCvU64+BpCYhtucqdN8FcE+FtGNCSJ7Sv7ejPOlbHQ43SN06kQKBgQCo0HRsEbp1Fx8EBmyVIxtcdR0lCvsPZJ4TPu6gJOkq1bRwMKTTyKPFwX7SvFT9Mcj6x4UsEE1ahSQLMyxIbokJ4s0Bkpyy3Ebo3eTha6b9D8BRNbDQL+orbgSaLLjn29oQmoC7TKyjN0jyyp0UMP5EhMJfiKXCCZzQdTrNlZa5iw=="
  },
```

Thanks in advance for any clues! I'm really excited to use deis and i hope that might be resolved :)
Mike

**Comment by carmstrong** [comment]
`sshKeyData` should be the *public* key material for your key, so in your example you should be specifying the contents of `~/.ssh/deis.pub` as you later said you tried.

However, `deisctl` will need your *private* key to actually access the host, so you'll `ssh-add` that key.

**Comment by mbajur** [comment]
Ok, so that's what i thought before and even tried but it gives me another issue. I'm not able to finish next step of the tutorial which is

    deisctl config platform set sshPrivateKey=~/.ssh/deis

This command gives me "Error: failed initializing SSH client: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain"

I've seen that in troubleshooting section and it states that i need to make sure my `deis` key is added to `ssh-add` - it is. I'm also able to directly ssh into that server and the `deis` key is being used properly.

***

ps. acording to the original question, as a evidence - here is an article that suggests that the private key should be used (note point 7): https://azure.microsoft.com/pl-pl/documentation/articles/virtual-machines-deis-cluster/

Basically, maybe that's obvious but few tutorials on the internet suggests it the same way as the one above so maybe it would be a good idea to include some more details (about using the private key) in the official deis azure tutorial? But still, i'm not any kind of a devop or sysadmin so that all might be too obvious

**Comment by mbajur** [comment]
Not sure if that might be helpfull but there are a lot of bad things going on in `journalctl`. A lot of log entries similar to:

```
Dec 21 15:55:33 deisNode0 unknown[18715]: stopping listening for peers on http://10.0.0.6:2380
Dec 21 15:55:33 deisNode0 unknown[18715]: member "4da456e4f05648669e99ef5588102a00" has previously registered with discovery service token (https://discovery.etcd.io/5f574bc884c3fb1184b228cbd3a8a135).
Dec 21 15:55:33 deisNode0 unknown[18715]: But etcd could not find valid cluster configuration in the given data dir (/var/lib/etcd2).
Dec 21 15:55:33 deisNode0 unknown[18715]: Please check the given data dir path if the previous bootstrap succeeded
Dec 21 15:55:33 deisNode0 unknown[18715]: or use a new discovery token if the previous bootstrap failed.
Dec 21 15:55:33 deisNode0 systemd[1]: etcd2.service: Main process exited, code=exited, status=1/FAILURE
Dec 21 15:55:33 deisNode0 systemd[1]: Failed to start etcd2.
Dec 21 15:55:33 deisNode0 systemd[1]: etcd2.service: Unit entered failed state.
Dec 21 15:55:33 deisNode0 systemd[1]: etcd2.service: Failed with result 'exit-code'.
Dec 21 15:55:33 deisNode0 sdnotify-proxy[18634]: E1221 15:55:33.586890 00001 network.go:53] Failed to retrieve network config: client: etcd cluster is unavailable or misconfigured
Dec 21 15:55:34 deisNode0 sudo[18721]: azureuser : TTY=pts/0 ; PWD=/home/azureuser ; USER=root ; COMMAND=/bin/journalctl -f
Dec 21 15:55:34 deisNode0 sdnotify-proxy[18634]: E1221 15:55:34.587551 00001 network.go:53] Failed to retrieve network config: client: etcd cluster is unavailable or misconfigured
Dec 21 15:55:35 deisNode0 sdnotify-proxy[18634]: E1221 15:55:35.588557 00001 network.go:53] Failed to retrieve network config: client: etcd cluster is unavailable or misconfigured
Dec 21 15:55:36 deisNode0 sdnotify-proxy[18634]: E1221 15:55:36.589581 00001 network.go:53] Failed to retrieve network config: client: etcd cluster is unavailable or misconfigured
Dec 21 15:55:37 deisNode0 sdnotify-proxy[18634]: E1221 15:55:37.590508 00001 network.go:53] Failed to retrieve network config: client: etcd cluster is unavailable or misconfigured
Dec 21 15:55:38 deisNode0 sdnotify-proxy[18634]: E1221 15:55:38.591780 00001 network.go:53] Failed to retrieve network config: client: etcd cluster is unavailable or misconfigured
Dec 21 15:55:39 deisNode0 sdnotify-proxy[18634]: E1221 15:55:39.592805 00001 network.go:53] Failed to retrieve network config: client: etcd cluster is unavailable or misconfigured
Dec 21 15:55:40 deisNode0 sdnotify-proxy[18634]: E1221 15:55:40.593616 00001 network.go:53] Failed to retrieve network config: client: etcd cluster is unavailable or misconfigured
Dec 21 15:55:41 deisNode0 sdnotify-proxy[18634]: E1221 15:55:41.594392 00001 network.go:53] Failed to retrieve network config: client: etcd cluster is unavailable or misconfigured
Dec 21 15:55:42 deisNode0 sdnotify-proxy[18634]: E1221 15:55:42.595541 00001 network.go:53] Failed to retrieve network config: client: etcd cluster is unavailable or misconfigured
Dec 21 15:55:42 deisNode0 systemd[1]: flanneld.service: Start operation timed out. Terminating.
Dec 21 15:55:42 deisNode0 dockerd[926]: time="2015-12-21T15:55:42.948489284Z" level=info msg="POST /v1.20/containers/b806f254b93a66fe8741ec996c9d2426bc0f52ed56fb60ef36eb90b7b3a1a99f/kill?signal=TERM"
Dec 21 15:55:42 deisNode0 dockerd[926]: time="2015-12-21T15:55:42.949667231Z" level=info msg="POST /v1.20/containers/b806f254b93a66fe8741ec996c9d2426bc0f52ed56fb60ef36eb90b7b3a1a99f/kill?signal=TERM"
Dec 21 15:55:42 deisNode0 sdnotify-proxy[18634]: I1221 15:55:42.951527 00001 main.go:292] Exiting...
```

And they are showing all the time, i'm not doing anything particular on the VPS, just `journalctl -f`.

**Comment by monaka** [comment]
@mbajur I guess the public key should be same as the content in *.pub.
Like `ssh-rsa AAAA{snip}wszl deis`.
But you put just `AAAA{snip}wszl` only.

I'm not sure because I'm already deployed them with success. But I got a deployment failure similar to yours before.

**Comment by bacongobbler** [comment]
This is not something we'll be able to get to for the LTS release (#4776), and provisioning Deis v2 is done through installing a vanilla kubernetes cluster so I'm going to go ahead and close this.
